build/android/app/src/main/java/net/minetest/minetest/CopyZipTask.java

```diff
@@ -51,11 +52,13 @@ protected void onPostExecute(String result) {
 
 	private void copyAsset(String zipName) {
 		String filename = zipName.substring(zipName.lastIndexOf("/") + 1);
-		try (InputStream in = contextRef.get().getAssets().open(filename);
-		     OutputStream out = new FileOutputStream(zipName)) {
+		try (InputStream in = activityRef.get().getAssets().open(filename);
+			 OutputStream out = new FileOutputStream(zipName)) {
 			copyFile(in, out);
 		} catch (IOException e) {
-			Toast.makeText(contextRef.get(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show();
+			if (activityRef.get() != null) {
+				activityRef.get().runOnUiThread(() -> Toast.makeText(activityRef.get(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show());
```

Comment:
no, you need to store this in a local variable. It's possible the activity will die in between the get calls

Reply:
I am afraid activity reference would be invalid in this case and toast won't appear anyway. It is still a valid point so I will do it.